src/main/scala/nl/biopet/tools/bamstats/generate/Generate.scala

```diff
@@ -55,8 +55,8 @@ object Generate extends ToolCommand[Args] {
           getDictFromBam(cmdArgs.bamFile)
       }
     val samReader = SamReaderFactory.makeDefault().open(cmdArgs.bamFile)
-    val stats: GroupStats = cmdArgs.bedFile match {
-      case Some(bed: File) if cmdArgs.onlyUnmapped =>
+    val root: BamstatsRoot = cmdArgs.bedFile match {
```

Comment:
readgroup parsing now happens inside the regions, should be done only once before touching the reads/iterators